test/validator.test.ts

```diff
@@ -153,7 +153,172 @@ describe('Validator', () => {
     };
 
     const result = validator.validate(node, natspec);
-    expect(result).toContainEqual(`@return missing for unnamed return`);
+    expect(result).toContainEqual(`@return missing for unnamed return №2`);
+  });
+
+  it('should warn of missed unnamed return', () => {
+    node = contract.vFunctions.find(({ name }) => name === 'externalSimpleMultipleUnnamedReturn')!;
+    let natspec = {
+      tags: [
+        {
+          name: 'notice',
+          content: 'External function that returns a bool',
+        },
+        {
+          name: 'dev',
+          content: 'A dev comment',
+        },
+      ],
+      params: [],
+      returns: [
+        {
+          name: 'Some',
+          content: 'return data',
+        },
+      ],
+    };
+
+    const result = validator.validate(node, natspec);
+    expect(result).toEqual([`@return missing for unnamed return №2`]); // only 1 warning
+  });
+
+  it('should not warn of extra natspec tags', () => {
+    node = contract.vFunctions.find(({ name }) => name === 'externalSimpleMultipleUnnamedReturn')!;
+    let natspec = {
+      tags: [
+        {
+          name: 'notice',
+          content: 'External function that returns a bool',
+        },
+        {
+          name: 'dev',
+          content: 'A dev comment',
+        },
+      ],
+      params: [],
+      returns: [
+        {
+          name: 'Some',
+          content: 'return data',
+        },
+        {
+          name: 'Some',
+          content: 'return data',
+        },
+      ],
+    };
+
+    const result = validator.validate(node, natspec);
+    expect(result).toEqual([]); // no warnings
+  });
+
+  it('should warn if tags are in a wrong order', () => {
+    node = contract.vFunctions.find(({ name }) => name === 'externalSimpleMultipleReturn')!;
+    let natspec = {
+      tags: [
+        {
+          name: 'notice',
+          content: 'External function that returns a bool',
+        },
+        {
+          name: 'dev',
+          content: 'A dev comment',
+        },
+      ],
+      params: [
+        {
+          name: '_magicNumber',
+          content: 'A parameter description',
+        },
+        {
+          name: '_name',
+          content: 'Another parameter description',
+        },
+      ],
+      returns: [
+        {
+          name: 'Some',
+          content: 'return data',
+        },
+        {
+          name: '_isMagic',
+          content: 'Some return data',
+        },
+      ],
+    };
+
+    const result = validator.validate(node, natspec);
+    expect(result).toEqual(['@return _isMagic is missing']); // 1 warning
```

Comment:
The warning is misleading here. As a user, I will check that the `@return _isMagic` is there and think that the tool has mixed something up. Some will realize it's because of the order but many won't. We should either make it the message or avoid printing any warnings in case of incorrect ordering.

Reply:
> The warning is misleading here. As a user, I will check that the `@return _isMagic` is there and think that the tool has mixed something up. Some will realize it's because of the order but many won't. We should either make it the message or avoid printing any warnings in case of incorrect ordering.

this won't compile as well so I'll delete it